spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/HttpClientProperties.java

```diff
@@ -326,6 +387,54 @@ public void setTrustedX509Certificates(List<String> trustedX509) {
 			}
 		}
 
+		public KeyManagerFactory getKeyManagerFactory() {
```

Comment:
I wish we wouldn't have let `getTrustedX509CertificatesForTrustManager()` into this class. I think that these should be moved to `GatewayAutoConfiguration`.

Reply:
This might be done in order to have all the ssl related config in one class and have other class(GatewayAutoConfiguration) utilize it